ds3-cli-certification/src/test/java/com/spectralogic/ds3cli/certification/Certification_Test_7.java

```diff
@@ -0,0 +1,454 @@
+        boolean success = false;
+        try {
+            Util.command(invalid_client, "--http -c get_service");
+        } catch(final UnknownHostException uhe) {
+            final String formattedException = ExceptionFormatter.format(uhe);
+            final String expectedError = "UnknownHost";
+            assertThat(formattedException, containsString(expectedError));
+            OUT.insertLog("CommandResponse for 7.2.2 invalid endpoint:");
+            OUT.insertPreformat(formattedException);
+            success = true;
+        } finally {
+            OUT.finishTest(testDescription, success);
+        }
+    }
+
+    @Test
+    public void test_7_3_1_list_nonexistent_bucket() throws Exception {
+        final String testDescription = "7.3.1: Non-extant bucket";
+        final String bucketName = CertificationUtil.getBucketName(testDescription);
+        OUT.startNewTest("");
+        boolean success = false;
+
+        try {
+            Util.command(client, "--http -c get_bucket -b " + bucketName);
+        } catch(final CommandException ce) {
+            final String formattedException = ExceptionFormatter.format(ce);
+            assertThat(formattedException, containsString("Error: Unknown bucket."));
+            OUT.insertLog("CommandResponse for 7.3.1 failed attempt to list nonexistent bucket:");
+            OUT.insertPreformat(formattedException);
+            success = true;
+        } finally {
+            OUT.finishTest(testDescription, success);
+        }
+    }
+
+    @Test
+    public void test_7_3_2_access_bucket_wrong_user() throws Exception {
+        final String testDescription = "7.3.2: Access bucket wrong user";
+        final String bucketName = CertificationUtil.getBucketName(testDescription);
+        OUT.startNewTest(testDescription);
+        boolean success = false;
+        try {
+            // Create a bucket
+            final String createBucketCmd = "--http -c put_bucket -b " + bucketName;
+            final CommandResponse createBucketResponse = Util.command(client, createBucketCmd);
+            assertThat(createBucketResponse.getReturnCode(), is(0));
+
+            // Create a new user, and wrap it with a new client
+            OUT.insertLog("Create user " + NO_RIGHTS_USERNAME);
+            CertificationUtil.createUser(client, NO_RIGHTS_USERNAME);
+            final String noRightsUserSecretKey = CertificationUtil.getUserSecretKey(client, NO_RIGHTS_USERNAME);
+            final Credentials badCreds = new Credentials(NO_RIGHTS_USERNAME, noRightsUserSecretKey);
+            final String endpoint = System.getenv("DS3_ENDPOINT");
+            final Ds3Client invalid_client = Ds3ClientBuilder.create(endpoint, badCreds).withHttps(false).build();
+
+            // Attempt to access the bucket with the new user, which should fail
+            final String listBucketCmd = "--http -c get_bucket -b " + bucketName;
+            Util.command(invalid_client, listBucketCmd);
```

Comment:
This should have a fail() statement after the Util.command, otherwise you'll get a false positive